tck/cmd/tck_shoppingcart/shoppingcart.go

```diff
@@ -65,15 +66,8 @@ type ShoppingCart struct {
 	cloudstate.EventEmitter
 }
 
-// New implements EntityInitializer and returns a new
-// and initialized instance of the ShoppingCart entity.
-func (sc ShoppingCart) New() interface{} {
-	return NewShoppingCart()
-}
-
-// NewShoppingCart returns a new and initialized
-// instance of the ShoppingCart entity.
-func NewShoppingCart() *ShoppingCart {
+// NewShoppingCart returns a new and initialized instance of the ShoppingCart entity.
+func NewShoppingCart() cloudstate.Entity {
 	return &ShoppingCart{
 		cart:         make([]*domain.LineItem, 0),
 		EventEmitter: cloudstate.NewEmitter(), // TODO: the EventEmitter could be provided by the event sourced handler
```

Comment:
Are we still planning to make this change? So it is passed on the context similar to the java library.

Reply:
Yes we do. I have to think about how to do it and iterate on it.